### ISSUE TITLE
Right button click on a plot with profile tools leads to an exception

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#3335]
With both current master and 0.14 branch, creating a horizontal line profile on an image and then pressing right click on the plot leads to this exception:
```
ERROR:silx.gui.qt._qt:<class 'TypeError'> 'float' object is not subscriptable   File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
    if roi.contains(data):
  File "/silx/gui/plot/items/roi.py", line 405, in contains
    return position[1] == self.getPosition()[1]
```
It is not the case with silx v0.13.2.

Reproducible with `examples/plotInteractiveImageROI.py` or `silx view` with an image

**Comment by t20100** [comment]
Another one with free line profile:
```
Traceback (most recent call last):
  File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
    if roi.contains(data):
  File "/silx/gui/plot/items/roi.py", line 321, in contains
    line_pt1 = self._points[0]
AttributeError: 'ProfileImageLineROI' object has no attribute '_points'
```

**Comment by t20100** [comment]
A few more with `examples/plotInteractiveImageROI.py` create a ROI of the given type and right click on the plot:

- With point ROI:
```
Traceback (most recent call last):
  File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
    if roi.contains(data):
  File "/silx/gui/plot/items/roi.py", line 120, in contains
    raise NotImplementedError('Base class')
NotImplementedError: Base class
```
- With circle ROI:
```
Traceback (most recent call last):
  File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
    if roi.contains(data):
  File "/silx/gui/plot/items/_roi_base.py", line 496, in contains
    raise NotImplementedError("Base class")
NotImplementedError: Base class

```
- With ellipse ROI:
```
Traceback (most recent call last):
  File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
    if roi.contains(data):
  File "/silx/gui/plot/items/_roi_base.py", line 496, in contains
    raise NotImplementedError("Base class")
NotImplementedError: Base class
```

Looks like a cluster.

**Comment by payno** [comment]
the contains was used for stats. But I didn't expected it to be call from something else... strange

**Comment by payno** [comment]
> Another one with free line profile:
> 
> ```
> Traceback (most recent call last):
>   File "/silx/gui/plot/tools/roi.py", line 606, in _feedContextMenu
>     if roi.contains(data):
>   File "/silx/gui/plot/items/roi.py", line 321, in contains
>     line_pt1 = self._points[0]
> AttributeError: 'ProfileImageLineROI' object has no attribute '_points'
> ```

For this one I guess the API has change. It could come from merging stats PR to the master. But the PR was old

**Comment by t20100** [comment]
I fixed the first 2 reported (and a similar one with vertical line profile).
I am fixing the points, circle, ellipse and make the default more resilient.

There is a second issue: this `contains` is used for checking if right click occurs over a ROI, but `contains` check is exact (i.e., there is no tolerance around the position), so it pretty much always return `False` for points or lines since the mouse is almost never exactly at the right position.

**Comment by payno** [comment]
Yes but the 'exact check' is required for stats. I guess we need two functions or an option.

**Comment by t20100** [comment]
OK, for now, I am keeping the checks as they are, but if it is used for interaction it is always best to have a tolerance of a few pixels around the mouse.

**Comment by vallsv** [comment]
Nice to catch it now.

**Comment by vallsv** [comment]
Yes i did use the `contains` as a trick to check the mouse click location.

The default to `False` done by @t20100 is a nice change.

Sorry, i did not notice it was not always implemented.

**Comment by t20100** [comment]
closed by PR #3336 on branch 0.14
